controllers/mattermost/mattermost/controller.go

```diff
@@ -144,10 +144,18 @@ func (r *MattermostReconciler) Reconcile(ctx context.Context, request ctrl.Reque
 		return reconcile.Result{}, err
 	}
 
-	err = r.checkMattermost(mattermost, dbConfig, fileStoreConfig, &status, reqLogger)
-	if err != nil {
-		r.updateStatusReconcilingAndLogError(mattermost, status, reqLogger, err)
-		return reconcile.Result{}, err
+	recStatus := reconcileStatus{
+		Status: false,
+		Error:  nil,
+	}
+
+	recStatus = r.checkMattermost(mattermost, dbConfig, fileStoreConfig, &status, reqLogger)
+	if recStatus.Error != nil {
+		if recStatus.Status {
+			time.Sleep(5 * time.Second)
+		}
+		r.updateStatusReconcilingAndLogError(mattermost, status, reqLogger, recStatus.Error)
+		return reconcile.Result{}, recStatus.Error
 	}
```

Comment:
We do not need to sleep here, we can requeue the resource after some time instead:
```suggestion
	recStatus, err := r.checkMattermost(mattermost, dbConfig, fileStoreConfig, &status, reqLogger)
	if err != nil {
		r.updateStatusReconcilingAndLogError(mattermost, status, reqLogger, recStatus.Error)
		return reconcile.Result{}, recStatus.Error
	}
	if !recStatus.ResourceReady {
	    // TODO: add some log here
	    return ctrl.Result{RequeueAfter: 10*time.Second}, nil // TODO: extract duration as constant
	}
```